src/packaging/requirements.py

```diff
@@ -38,7 +38,7 @@ def __init__(self, requirement_string: str) -> None:
 
         self.name: str = parsed.name
         self.url: Optional[str] = parsed.url or None
-        self.extras: Set[str] = set(parsed.extras if parsed.extras else [])
+        self.extras: Set[str] = set(parsed.extras or [])
```

Comment:
Similar to `parsed.url or None ` in the previous line. I find consistency in this specific case improves readability.